packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5890,47 +5894,134 @@ class MouseRegion extends SingleChildRenderObjectWidget {
+  /// mounted, an exception will occur. This is because the callback is
+  /// triggered during the post-frame phase, at which point the widget has been
+  /// unmounted. Since [setState] is exclusive to widgets, the restriction is
+  /// specific to [MouseRegion], and does not apply to its lower-level
+  /// counterparts, [RenderMouseRegion] and [MouseTrackerAnnotation].
+  ///
+  /// There are a few ways to mitigate this restriction:
+  ///
+  ///  * If the hover state is completely contained within a widget that
+  ///    unconditionally creates this [MouseRegion], then this will not be a
+  ///    concern, since after the [MouseRegion] is unmounted the state is no
+  ///    longer used.
+  ///  * Otherwise, the outer widget very likely has access to the variable that
+  ///    controls whether this [MouseRegion] is present. If so, call [onExit] at
+  ///    the event that turns the condition from true to false.
+  ///  * In cases where the solutions above won't work, you can always
+  ///    override [State.dispose] and call [onExit], or create your own widget
+  ///    using [RenderMouseRegion].
+  ///
+  /// {@tool snippet}
+  /// The following example shows a stateful widget where the hover state is
+  /// completely contained within a widget that unconditionally creates a
+  /// `MouseRegion`. In this case, you can ignore this restriction.
+  ///
+  /// ```dart
+  /// class MyButton extends StatefulWidget {
+  ///   @override
+  ///   State<StatefulWidget> createState() => _MyButtonState();
+  /// }
+  ///
+  /// class _MyButtonState extends State<MyButton> {
+  ///   bool hovered = false;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return MouseRegion(
+  ///       onEnter: (_) { setState(() { hovered = true; }); }
+  ///       onExit: (_) { setState(() { hovered = false; }); }
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// The following example shows a stateful widget that conditionally creates a
+  /// `MouseRegion`, and leaks the hover state to outside. Since this widget has
+  /// access to the event that triggers the disappearance of the `MouseRegion`,
+  /// it should trigger the callback during that event as well.
+  ///
+  /// ```dart
+  /// // MyListenerButton listens to the network and hides its MouseRegion if told
+  /// // so. It also leaks the exit event to its parent.
+  /// class MyListenerButton extends StatefulWidget {
+  ///   MyListenerButton({ this.onEnterButton, this.onExitButton });
+  ///
+  ///   final VoidCallback onEnterButton;
+  ///   final VoidCallback onExitButton;
+  ///
+  ///   @override
+  ///   State<StatefulWidget> createState() => _MyListenerButtonState();
+  /// }
+  ///
+  /// class _MyListenerButtonState extends State<MyListenerButton> {
+  ///   final bool hideButton = false;
+  ///   final bool hovered = false;
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return MyNetworkListener(
```

Comment:
I think this will fail the samples analysis, since `MyNetworkListener` isn't a defined class. You can either define it (which would be preferred), or declare it at the top of the file in a "// Examples can assume:" block so that it is found, even if it isn't actually functional.  If you can think of a simple example that can be coded here and doesn't use the network, that would probably be best.

See here for more information: https://github.com/flutter/flutter/tree/master/dev/snippets#analysis

Reply:
I've turned it into a full application sample.